### DISCUSSION TITLE
.p8 content into .env for CI - Security

### DISCUSSION THREAD

**Discussion by Parigi65** [fastlane/fastlane#17473] [Q&A] [answered]
cc: @joshdholtz 

I'm reading the https://github.com/fastlane/fastlane/issues/17334 to resolve an issue of mine and I read this comment:
https://github.com/fastlane/fastlane/pull/17061#issuecomment-698565071
that seems to be a solution for me, but I have to ask a question about the security of this procedure:

Bamboo need to checkout the project code before perform any task like fastlane, so, I have to put the .env file into the repo. Otherwise fastlane can't work, am i wrong?

If I put the .env file into the repository, everyone who have permissions on the repo can take my .env file and use it for other applications or other Apple oriented operations.

Are my thoughts correct?

**Comment by rogerluan** [comment, marked as answer]
Do not commit your env vars or secrets to code versioning system! 🙏 

Your assumption is right - anyone will be able to use those. Env vars should be set via CI configuration settings somewhere in their dashboard (probably). Not via git.

See this for further instructions: https://confluence.atlassian.com/bamboo/variables-for-deployment-environments-342754180.html

**Reply by joshdholtz** [reply]
@Parigi65 This is correct ☝️  You should use whatever environment variable configuration tools that are built into your CI so they are stored and loaded securely.
